### PR TITLE
Add test for sign_tx()

### PR DESCRIPTION
**Pull request by wszdexdrf**
Also made necessary changes to automation.json (for ledger devices). Add
deserializing method for PSBT.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -145,10 +149,62 @@ mod tests {
...
+        let client = HWIClient::get_client(device, true, types::HWIChain::Test).unwrap();
+        let derivation_path = DerivationPath::from_str("m/44'/1'/0'/0/0").unwrap();
+
+        let address = client
+            .display_address_with_path(&derivation_path, types::HWIAddressType::Legacy)
+            .unwrap();
+
+        let pk = client.get_xpub(&derivation_path, true).unwrap();
+        let mut hd_keypaths: BTreeMap<secp256k1::PublicKey, KeySource> = Default::default();
+        hd_keypaths.insert(pk.public_key, (device.fingerprint, derivation_path));
```
Comment (danielabrozzoni) [inline src/lib.rs:166]: You should put the master xpub fingerprint in the map, which in this case it's the same as the device fingerprint. It's ok to leave it like that, but please add a small comment explaining

File: src/types.rs [hunk 2]
```diff
@@ -49,9 +49,19 @@ pub struct HWIAddress {
 
 #[derive(Clone, Eq, PartialEq, Debug, Deserialize)]
 pub struct HWIPartiallySignedTransaction {
+    #[serde(deserialize_with = "deserialize_psbt")]
```
Comment (danielabrozzoni) [inline src/types.rs:52]: Why is this needed? PSBT already has the deserialize method https://docs.rs/bitcoin/latest/bitcoin/util/psbt/struct.PartiallySignedTransaction.html#impl-Deserialize%3C%27de%3E

**Review by danielabrozzoni — COMMENTED** [review]
Concept ACK, a few comments.

Once again, stop putting everything in a single commit! This PR should have been separated in 3 commits: one for deserialize, one for the automation.json, one for the test itself.

File: src/types.rs (hunk 2 above)
Comment (wszdexdrf) [reply src/types.rs:52]: I agree that it shouldn't be needed, but it doesn't work without it. Serde complains that `expected struct PartiallySignedTransaction while deserializing`.

**Comment by wszdexdrf** [comment]
> Once again, stop putting everything in a single commit! 

I apologize for being lazy. I will keep it in mind. 🙇

File: src/types.rs (hunk 2 above)
Comment (danielabrozzoni) [reply src/types.rs:52]: Mh, weird. Ok to leave it like that then

File: src/lib.rs [hunk 3]
```diff
@@ -145,10 +149,63 @@ mod tests {
...
+                vout: Default::default(),
+            },
+            ..Default::default()
+        };
+        let psbt = bitcoin::psbt::PartiallySignedTransaction {
+            unsigned_tx: Transaction {
+                version: 1,
+                lock_time: 0,
+                input: vec![txin],
+                output: vec![],
```
Comment (danielabrozzoni) [inline src/lib.rs:191]: Here you should put a txout

File: src/lib.rs [hunk 4]
```diff
@@ -145,10 +149,63 @@ mod tests {
...
+            version: 1,
+            lock_time: 0,
+            input: vec![TxIn::default()],
+            output: vec![TxOut {
+                value: 100,
+                script_pubkey: address.address.script_pubkey(),
+            }],
+        };
+
+        let txin = TxIn {
```
Comment (danielabrozzoni) [inline src/lib.rs:179]: I would call this "previous_txin", so it's easier to understand that it's not the current tx's txin

File: src/lib.rs [hunk 5]
```diff
@@ -145,10 +149,63 @@ mod tests {
...
+        let address = client
+            .display_address_with_path(&derivation_path, types::HWIAddressType::Legacy)
+            .unwrap();
+
+        let pk = client.get_xpub(&derivation_path, true).unwrap();
+        let mut hd_keypaths: BTreeMap<secp256k1::PublicKey, KeySource> = Default::default();
+        // Here device fingerprint is same as master xpub fingerprint
+        hd_keypaths.insert(pk.public_key, (device.fingerprint, derivation_path));
+
+        let tx = Transaction {
```
Comment (danielabrozzoni) [inline src/lib.rs:169]: Should be called previous_tx

File: src/lib.rs [hunk 6]
```diff
@@ -145,10 +149,63 @@ mod tests {
...
+            unsigned_tx: Transaction {
+                version: 1,
+                lock_time: 0,
+                input: vec![txin],
+                output: vec![],
+            },
+            xpub: Default::default(),
+            version: 0,
+            proprietary: BTreeMap::new(),
+            unknown: BTreeMap::new(),
```
Comment (danielabrozzoni) [inline src/lib.rs:196]: You can probably use default on those?
